docs/customization.rst

```diff
@@ -174,101 +197,257 @@ Configuration Options
+                  |lightbulb-outline| ``material/lightbulb-outline``, |lightbulb| ``material/lightbulb``
+
+        .. confval:: media (palette preference)
+
+            In order to automatically set the color palette based on the user's system preference, a media
+            query can be specified with the ``media`` field.
+
+            .. code-block:: python
+
+                html_theme_options = {
+                    "palette": [
+                        {
+                            "media": "(prefers-color-scheme: light)",
+                            "scheme": "default",
+                            "toggle": {
+                                "icon": "material/toggle-switch-off-outline",
+                                "name": "Switch to dark mode",
+                            }
+                        },
+                        {
+                            "media": "(prefers-color-scheme: dark)",
+                            "scheme": "slate",
+                            "toggle": {
+                                "icon": "material/toggle-switch",
+                                "name": "Switch to light mode",
+                            }
+                        },
+                    ]
+                }
 
-            Options must be `any of the icons bundled with this theme <https://github.com/squidfunk/mkdocs-material/tree/master/material/.icons>`_.
-            Popular combinations are
+    .. confval:: direction
 
-            .. csv-table::
+        Specifies the text direction. Set to ``ltr`` (default) for left-to-right,
+        or ``rtl`` for right-to-left.
 
-                |toggle-off| ``material/toggle-switch-off-outline``, |toggle-on| ``material/toggle-switch``
-                |sunny| ``material/weather-sunny``, |night| ``material/weather-night``
-                |eye-outline| ``material/eye-outline``, |eye| ``material/eye``
-                |lightbulb-outline| ``material/lightbulb-outline``, |lightbulb| ``material/lightbulb``
+    .. confval:: font
 
-    .. confval:: direction
+        Use this dictionary to change the fonts used in the theme. For example:
 
-        Specifies the text direction.  Set to ``ltr`` (default) for left-to-right,
-        or ``rtl`` for right-to-left.
+        .. code-block:: python
+
+            html_theme_options = {
+                "font": {
+                    "text": "Roboto",  # used for all the pages' text
+                    "code": "Roboto Mono"  # used for literal code blocks
+                },
+            }
+
+        You can specify any of the available `Google Fonts <https://fonts.google.com/>`_.
 
     .. confval:: google_analytics_account
 
         Set to enable google analytics.
 
-    .. confval:: globaltoc_depth
+    .. confval:: globaltoc_collapse
 
-        The maximum depth of the global TOC; set it to -1 to allow unlimited depth.
+        If true (the default value), TOC entries that are not ancestors of the current page are collapsed.
```

Comment:
Consider adding a warning that setting this to `false` may lead to large generated page sizes since the entire table of contents tree will be duplicated on every page.